src/macros.jl

```diff
@@ -327,12 +327,12 @@ function _macro_assign_and_return(code, variable, name;
     end
 end
 
-function _check_vectorized(sense::Symbol)
-    sense_str = string(sense)
+function _check_vectorized(sense::Val{S}) where S
```

Comment:
Do you have an example for which the heuristic "vectorized iff `S` starts by a `.`" does not work ?

Reply:
I don't, for now, but you may have some with more developments of constraint programming, like `alldifferent.(vectorA, vectorB)`. I don't think we need to care that much about this case for now.

Reply:
Then why not keep `Symbol` ? Using `Val` means that you want to allow it to be extended.

Reply:
I had problems at some point and had to use Val (or do tricks I didn't know at the time). I've reverted to symbols.